ext/liquid_c/block.c

```diff
@@ -55,7 +56,15 @@ static VALUE rb_block_parse(VALUE self, VALUE tokens, VALUE options)
             }
             case TOKEN_RAW:
             {
-                VALUE str = rb_enc_str_new(token.str, token.length, utf8_encoding);
+                const char *start = token.str, *end = token.str + token.length;
+                const char *token_start = start;
+
+                if(token.lstrip)
+                    token_start = read_while(start, end, rb_isspace);
+
+                VALUE str = rb_enc_str_new(token_start, end - token_start, utf8_encoding);
+                if(token.rstrip)
+                    rb_funcall(str, intern_rstrip, 0);
```

Comment:
Yeah I actually was thinking to do exactly that. I really only stuck to this while refactoring so I could confirm i didn't introduce anything new. Should have some time tonight to pull this out and bring it up to where the lstrip sits.